chargeOclock/Source/Other/NetworkAPI.swift

```diff
@@ -43,28 +43,23 @@ class NetworkAPI {
 		guard let url = constructURL(for: uri, completion: completion) else {
 			return
 		}
-		URLSession.shared.dataTask(with: url) { data, response, error in
-			guard !self.isError(error, completion: completion) else {
-				return
-			}
-			guard let response = response as? HTTPURLResponse, self.checkResponseCode(response.statusCode, completion: completion) else {
-				return
-			}
-			guard let data = data else {
-				completion(nil, oClockError.serverError(code: response.statusCode))
-				return
-			}
-			completion(data, nil)
-			}.resume()
+
+		var urlRequest = URLRequest(url: url)
+		urlRequest.httpMethod = "GET"
+		performRequest(with: urlRequest, completion: completion)
 	}
 
 	fileprivate func performDeleteRequest(to uri: String, completion: @escaping NetworkCompletion) {
 		guard let url = constructURL(for: uri, completion: completion) else {
 			return
 		}
+
 		var urlRequest = URLRequest(url: url)
 		urlRequest.httpMethod = "DELETE"
+		performRequest(with: urlRequest, completion: completion)
+	}
 
+	private func performRequest(with urlRequest: URLRequest, completion: @escaping NetworkCompletion) {
```

Comment:
yes...
`performRequest(to url: URL, method httpMethod: String, completion: @escaping NetworkCompletion)`